src/hotspot/share/adlc/adlc.hpp

```diff
@@ -39,33 +39,6 @@
 #include <stdarg.h>
 #include <sys/types.h>
 
-/* Make sure that we have the intptr_t and uintptr_t definitions */
-#ifdef _WIN32
```

Comment:
As this is a synonym for `_WINDOWS` it is not obvious this deletion is correct.

Reply:
The deletion is apparently working, else we'd be getting build failures. So
while there are some potential issues and opportunities for further cleanup in
this file, I think they ought to be addressed separately from this PR. See
new https://bugs.openjdk.org/browse/JDK-8343530.

Reply:
There is a difference between "working" and not causing a build failure. I suspect none of that code is actually needed these days, but I'm not sure. As deleting the entire section goes beyond deleting 32-bit code, I would expect it to be partially restored in this PR and then cleaned up in a later PR.

Reply:
"using namespace std;" in a header is generally a bad idea. It brings all
kinds of stuff into scope, potentially leading to name conflicts down the
road.  And seems like a strange thing to do only for windows.

Removal of the strdup macro is covered by the NONSTDC macros added at build
time.  It's not a 32bit cleanup either, and you suggested it.

Removal of [u]intptr_t definitions will cause a build failure if it results in
them being undefined. And getting an incorrect definition from elsewhere seems
implausible. I claim this all just isn't needed anymore and can be removed in
this PR, just as you suggested for the strdup macro. The 64bit definitions
could be added back in this PR (to be removed later by JDK-8343530), but that
just seems like useless churn.

So I'm happy with the current removal of the entire chunk.

Reply:
It was already quite tangential to the 32-bit windows removal effort. I've restored the original code (with the exception of the 32-bit Windows part) in this PR, so we don't have to argue about that. 

Let's remove this as a separate effort, presumably as part of JDK-8343530.